var/spack/repos/builtin/packages/pastix/package.py

```diff
@@ -0,0 +1,89 @@
+    url      = "https://gitlab.inria.fr/solverstack/pastix/uploads/baa57033d98378e0f3affbf45900fb6e/pastix-6.2.1.tar.gz"
+    git      = "https://gitlab.inria.fr/solverstack/pastix.git"
+    maintainers = ['fpruvost', 'mfaverge', 'ramet']
+
+    version('master', branch='master', submodules=True)
+    version('6.2.1', '50742cc0e6e03728f7529fc607d5c65b9e14205f192947678d6103a872a6210c')
+
+    # cmake's specific
+    variant('shared', default=True, description='Build Pastix as a shared library')
+
+    # pastix's specific
+    variant('doc', default=False, description='Enable documentation')
+    variant('int64', default=False, description='To use 64 bits integers')
+    variant('metis', default=False, description='Enable Metis')
+    variant('scotch', default=True, description='Enable Scotch')
+    variant(
+        'runtime', default='none', description='Runtime support',
+        values=('none', 'starpu'), multi=False
+    )
+    variant('cuda', default=False, when='runtime=starpu', description='Enable CUDA')
+    variant('mpi', default=False, description='Enable MPI')
+
+    # Dependencies
+    depends_on("pkg-config", type='build')
+    depends_on("hwloc")
+    depends_on("lapack")
+    with when("+metis"):
+        depends_on("metis@5.1:")
+        depends_on("metis@5.1:+int64", when='+int64')
+    with when("+scotch"):
+        depends_on("scotch~metis")
+        depends_on("scotch~metis+int64", when='+int64')
+        depends_on("scotch~metis~mpi", when='~mpi')
+    with when("runtime=starpu"):
+        depends_on("starpu")
+        depends_on("starpu~mpi", when='~mpi')
+        depends_on("starpu+mpi", when='+mpi')
+        depends_on("starpu~cuda", when='~cuda')
+        depends_on("starpu+cuda", when='+cuda')
+        depends_on("cuda", when='+cuda')
+    depends_on("mpi", when='+mpi')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            "-Wno-dev",
+            self.define("CMAKE_COLOR_MAKEFILE", "ON"),
+            self.define("CMAKE_VERBOSE_MAKEFILE", "ON"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant("PASTIX_INT64", "int64"),
+            self.define_from_variant("PASTIX_ORDERING_METIS", "metis"),
+            self.define_from_variant("PASTIX_ORDERING_SCOTCH", "scotch"),
+            self.define("PASTIX_WITH_PARSEC", "OFF"),
+            self.define_from_variant("PASTIX_WITH_MPI", "mpi")
+        ]
+
+        if spec.satisfies('runtime=starpu'):
+            args.extend([self.define("PASTIX_WITH_STARPU", "ON")])
+            args.extend([self.define_from_variant("PASTIX_WITH_CUDA", "cuda")])
+
+        if ('^intel-mkl' in spec or '^intel-parallel-studio+mkl' in spec):
+            if ('threads=none' in spec):
+                args.extend([self.define("BLA_VENDOR", "Intel10_64lp_seq")])
+            else:
+                args.extend([self.define("BLA_VENDOR", "Intel10_64lp")])
```

Comment:
We should never link with a multi-threaded BLAS.

Reply:
Ok done